stubs/docutils/docutils/statemachine.pyi

```diff
@@ -1,3 +1,194 @@
-from _typeshed import Incomplete
+from collections.abc import Callable, Generator, Iterator, Sequence
+from re import Match, Pattern
+from typing import Any, Generic, TypeVar, overload
+from typing_extensions import Self, TypeAlias
 
-def __getattr__(name: str) -> Incomplete: ...
+_TransitionResult: TypeAlias = tuple[list[str], str | None, list[str]]
+_TransitionMethod: TypeAlias = Callable[[Match[str], list[str], str], _TransitionResult]
+
+class StateMachine:
+    input_lines: StringList | None
+    input_offset: int
+    line: str | None
+    line_offset: int
+    debug: bool
+    initial_state: str
+    current_state: str
+    states: dict[str, State]
+    observers: list[Callable[[Self], None]]
+    def __init__(self, state_classes: Sequence[type[State]], initial_state: str, debug: bool = False) -> None: ...
+    def unlink(self) -> None: ...
+    def run(
+        self,
+        input_lines: list[str] | StringList,
+        input_offset: int = 0,
+        context: list[str] | None = None,
```

Comment:
It says in the docstring that context is a "application-specific storage", and I only see it passed to states. Would it make more sense to use `Any` or even a type alias `_Context: TypeAlias = Any`?

Reply:
i did have it set up this way at one point. see https://github.com/python/typeshed/pull/11469#issuecomment-1975938313 and https://github.com/python/typeshed/pull/11469#issuecomment-1986788144

i could reinstate the generics

Reply:
Ok, sounds good then. You might still consider a type var (to `list[str]`) for documentation purposes, but I leave this decision to you.